storage/raft_transport.go

```diff
@@ -192,8 +192,18 @@ func (t *RaftTransport) processQueue(nodeID roachpb.NodeID) {
 		return
 	}
 	client := NewMultiRaftClient(conn)
+
+	var wg sync.WaitGroup
 	ctx, cancel := context.WithCancel(context.TODO())
-	defer cancel()
+	defer func() {
+		// After pending Raft requests have finished, cancel the Context to free
+		// gRPC resources.
+		wg.Wait()
+		if log.V(1) {
+			log.Infof("cancelling context")
+		}
+		cancel()
+	}()
```

Comment:
There are gRPC protobuf unmarshal errors that are causing this function to return when there are one or more in-flight snapshot requests. There's also an as-of-yet unexplained context cancellation that's affecting some snapshot requests, which also causes the function to return.